src/regexp.toit

```diff
@@ -1291,18 +1310,28 @@ class MiniExpMatch implements Match:
 
   start -> int: return registers_[first_capture_reg_]
 
+  start index/int -> int?:
+    if index > group_count: throw "OUT_OF_RANGE"
```

Comment:
Is it `> group_count` or `>= group_count` ?

And also check for negative numbers?
```suggestion
    if not 0 <= index <= group_count: throw "OUT_OF_RANGE"
```

Reply:
If there are 3 groups, then they are conventionally numbered 1,  2, and 3.  The 0th is the whole regexp.  I added the check for negative numbers.